pages/en/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -207,121 +205,192 @@ For a full list, see the <DocLink copy="SnarkyJS reference" url="/zkapps/snarkyj
 
 #### Smart Contract
 
-Let’s put this all together and create a basic smart contract. This where you define your custom logic.
+<!-- TODOs
+  IMO, needs to be fleshed out a bit more before sending users to "test your zkApp" or "advanced topics".
+  Basics that are missing:
+  * payments
+  * Mina.transaction / at least hinting at the basic mechanics of how to compile / prove
+  * permissions? (maybe an advanced topic)
+ -->
+
+Now that we have covered the basics of writing SnarkyJS programs, let's see how to create a smart contract.
 
 Smart contracts are written by extending the base class `SmartContract`:
 
 ```ts
 class HelloWorld extends SmartContract {}
 ```
 
-A smart contract can contain **on-chain state**, which is declared as a property
-on the class with the `@state` decorator:
-
-```ts
-class HelloWorld extends SmartContract {
-  @state(Field) x = State<Field>();
-}
-```
-
-Here, `x` is of type `Field`. Only types built out of Field can be used for state variables.
-In the current design, the state can consist of at most 8 Fields of 32 bytes each.
-States are initialized with the `State()` function.
-
 The `constructor` of a `SmartContract` is inherited from the base class and should not be overriden.
 It takes the zkApp account address (a public key) as its only argument:
 
 ```ts
 let zkAppKey = PrivateKey.random();
 let zkAppAddress = PublicKey.fromPrivateKey(zkAppKey);
 
-let zkApp = new SmartContract(zkAppAddress);
+let zkApp = new HelloWorld(zkAppAddress);
 ```
 
-##### SnarkyJS Deployment Method
+Later, we show you how you can _deploy_ a smart contract to an on-chain account.
+
+<Alert kind="note">
+  On Mina, there is no strong distinction between normal "user accounts" and
+  "zkApp accounts". A zkApp account is just a normal account that has a smart
+  contract deployed to it -- which essentially just means there's a verification
+  key stored on the account, which can verify off-chain execution proofs.
```

Comment:
`which can vefiy zero-knowledge proofs generated by the smart contract associated with this zkApp account`? 

Main thing is trying to change `off-chain execution proofs` to `zero-knowledge proofs` to not introduce new terminology that is different from what we used in the How zkApps Works section